hlx_statics/blocks/discoverblock/discoverblock.js

```diff
@@ -22,26 +22,34 @@ export default async function decorate(block) {
   const mainContainer = document.querySelector('section');
   const discoverWrappers = document.querySelectorAll('.discoverblock-wrapper');
   
-  const allDiscoverblock = document.createElement('div');
-  allDiscoverblock.classList.add('all-discoverblock');
-  
-  discoverWrappers.forEach(wrapper => {
-    allDiscoverblock.appendChild(wrapper);
-  });
-  
-  const firstHeading2Wrapper = document.querySelector('.heading2-wrapper');
-  if (firstHeading2Wrapper) {
-    firstHeading2Wrapper.insertAdjacentElement('afterend', allDiscoverblock);
-  } else {
-    mainContainer.appendChild(allDiscoverblock);
-  }
-  const existingAllDiscoverblocks = document.querySelectorAll('.all-discoverblock');
-  
-  existingAllDiscoverblocks.forEach((block, index) => {
-    if (index > 0) {
-      block.remove();
+  const arrange = block?.getAttribute('data-arrange');
```

Comment:
I'm seeing that this implementation relies on us adding the `arrange` attribute on the product team's markdown file: https://github.com/AdobeDocs/app-builder/commit/9429f7e9b968cade96c7051f84694240959fc8fe

Looking at aio-theme readme, this property didn't exist before: https://github.com/adobe/aio-theme?tab=readme-ov-file#discover-block

Is it possible to fix this page without us having to add that `arrange` attribute?

Reply:
@manimitrah , could you check with Baskar on why he changed `DiscoverBlock` to `Discover` here: https://github.com/AdobeDocs/adobe-dev-console/commit/460fe803ee63895b24a68a1092dca2ab4d1306bf

I'm wondering if instead of adding the `arrange` attribute, we can change `DiscoverBlock` to `Discover` for consistency across all the repos.

Reply:
@melissag-ensemble , The same "discover" block is used to display single columns in some cases and grouped items in others. That’s why I added the "arrange" prop. Based on my analysis, I’m not sure it’s possible to handle this without it.
![Screenshot 2024-11-29 114354](https://github.com/user-attachments/assets/534bc097-369b-4856-aded-98c81b3b2d14)

Reply:
I think it's the width that controls whether the block appears in single columns (width 100%) or multi columns (width < 100%). If width is not specified, the default is < 100% (https://github.com/adobe/aio-theme/blob/main/packages/gatsby-theme-aio/src/components/DiscoverBlock/index.js#L21)

For example:

**width=100%** - shows single columns
- md: https://github.com/AdobeDocs/app-builder/blob/060423d955204bdaffcff8d8267ffe2cd001f56b/src/pages/resources/videos/index.md?plain=1#L34
- page: https://developer.adobe.com/app-builder/docs/resources/videos/

**width="25%"** - shows multi columns
- md: https://github.com/AdobeDocs/cja-apis/blob/eeb4924be1143d0d9d8407ad07450f077e46b19f/src/pages/support.md?plain=1#L20
- page: https://developer.adobe.com/cja-apis/docs/support/

**width not specfied** - shows multi columns
- md: https://github.com/AdobeDocs/ff-services-docs/blob/5dbfec4c605895498914446e375afd444aee8c5c/src/pages/guides/get-started.md?plain=1#L170
- page: https://developer.adobe.com/firefly-services/docs/guides/get-started/

Reply:
@melissag-ensemble, I have handled this without using arrange and have removed it from app-builder.